### PR TITLE
Move storage options to separate input in read_text

### LINKED ISSUE #1237: `storage_options` argument to `dataframe.read_csv` is out of sync with `bag.read_text` and open_files methods
Since #1184 is not released yet, I think it is a good time to align interfaces of all `read_*` methods. Currently, you have to pass `protocol://path` string to the first (`path`) argument and, in case of `read_csv`, a dict with storage options as `storage_options` argument, while `bag.read_text` and `open_files` methods expect storage options in `**kwargs`.

I think `storage_options` should be just implemented everywhere instead of polluting `**kwargs`. Also, I suggest to parse `path` (first argument to `read_*` and `open_files` methods) with [`urllib.parse` (py3)](https://docs.python.org/3/library/urllib.parse.html) / [`urlparse` (py2)](https://docs.python.org/2/library/urlparse.html) instead of dummy splits.

This issue is related to #1232.

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #1237

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @frol

**Comment by frol** [comment]
@mrocklin I was about to push my version of the fix, which includes `urlparse` as well.

**Comment by frol** [comment]
Would you please wait a couple of minutes to review and decide which way is better?

**Comment by mrocklin** [comment]
Sure

**Comment by mrocklin** [comment]
@frol I'd like to issue a release of dask sometime soon.  Thoughts on merging this now and adding in the `urlparse` functionality in #1269 later?

Are any there changes in this PR with which you disagree?

File: dask/bag/text.py [hunk 1]
```diff
@@ -57,10 +57,12 @@ def read_text(path, blocksize=None, compression='infer',
     --------
     from_sequence: Build bag from Python sequence
     """
+    storage_options = storage_options or {}
     if isinstance(path, (tuple, list, set)):
         blocks = sum([read_text(fn, blocksize=blocksize,
                       compression=compression, encoding=encoding, errors=errors,
-                      linedelimiter=linedelimiter, collection=False, **kwargs)
+                      linedelimiter=linedelimiter, collection=False,
+                      **storage_options)
```
Comment (frol) [inline dask/bag/text.py]: Look, you have `storage_options` argument, but here you expand its values. It should be just `storage_options=storage_options`.

File: dask/bag/text.py [hunk 2]
```diff
@@ -77,13 +79,14 @@ def read_text(path, blocksize=None, compression='infer',
...
-                                          compression=compression, **kwargs)
+                                          compression=compression,
+                                          **storage_options)
             blocks = [delayed(list)(file) for file in files]
 
         else:
             _, blocks = read_bytes(path, delimiter=linedelimiter.encode(),
                     blocksize=blocksize, sample=False, compression=compression,
-                    **kwargs)
+                    **storage_options)
```
Comment (frol) [inline dask/bag/text.py]: Same story as above here.

**Comment by frol** [comment]
If it is not 1.0 release, then I would say it is fine to merge this now and improve things later.

File: dask/bag/text.py (hunk 1 above)
Comment (mrocklin) [inline dask/bag/text.py]: Can you say a bit about why this is preferred?  How does expanding out `storage_options` negatively affect things?

Comment (frol) [inline dask/bag/text.py]: @mrocklin Hmm, it does not "negatively affect", but rather just does not follow the function definition, so it just breaks things. Here is just a minimized example of what is happening now:

```python
def read_text(path, ..., storage_options=None):
    blocks = [read_text(fn, **storage_options) ...]
```

while it should be

```python
def read_text(path, ..., storage_options=None):
    blocks = [read_text(fn, ..., storage_options=storage_options) ...]
```

Comment (mrocklin) [inline dask/bag/text.py]: Ah I see.  Thanks.  Nice save.  It's a bit concerning that this didn't come up on tests.

Comment (mrocklin) [inline dask/bag/text.py]: Fixed and added test

File: dask/bag/text.py (hunk 2 above)
Comment (mrocklin) [inline dask/bag/text.py]: This one, I think, is still OK.  These are keyword arguments that `read_bytes` genuinely expects

Comment (frol) [inline dask/bag/text.py]: OK. Anyway, I have aligned this `storage_options` usage in all of the places in my PR.

**Pull request closed without merging**